File: DotNet/CodeSamplesForAssertion/ClassWithExceptions.cs
using System;

namespace CodeSamplesForAssertion
{
    public class ClassWithExceptions
    {
        //This could qualify for two exception tests. 1. Assert it throws the correct exception. 2. Assert correct exception message
        public string SayHello(string yourName)
        {
            if (yourName == null)
            {
                throw new ArgumentNullException(yourName, "Name cannot be null");
            }
            return "Hello " + yourName;

        }
    }
}

File: java/src/main/java/com/agile/CodeSamplesForAssertion/ClassWithExceptions.java
package com.agile.CodeSamplesForAssertion;


public class ClassWithExceptions {

    //This could qualify for two exception tests.
    //1. Assert it throws the correct exception. 2. Assert correct exception message
	public String sayHello(String name) throws Exception{
		if (name == null)
            {
                throw new NullPointerException("Name is Null");
            }
            return "Hello " + name;

	}

}
